### PR TITLE
Enable system VM volume migration for KVM

### PR DESCRIPTION
**Pull request by GutoVeronezi**
### Description

Release 4.16.0.0 introduced a feature for migrating system VM volumes (#4385). However, it was enabled only for VMWare.

This PR intends to enable the feature for KVM too.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ x] Minor

### How Has This Been Tested?

It was tested in a local lab.

(I tested the process for each system VM type)

To test the SSVM and CPVM, I enabled a zone to create them. When they were created, I disabled the zone. I accessed the system VMs and created files to validate later. I stopped them and migrated the volumes. I checked the references in database and storage and then I started the system VMs. After that, I accessed them again and verified the files. The migration performed successfully, the data was kept and the system VMs were working fine.

To test the VR, I created a network. I accessed the VR and created a file to validate later. I stopped the VR and migrated the volume. I checked the reference in database and storage and then I started the VR. After that, I accessed it again and verified the file. The migration performed successfully, the data was kept and the VR was working fine.

### CODE REVIEW

**Comment by rohityadavcloud** [comment]
Can you merge latest main branch to your PR branch @GutoVeronezi ?

**Comment by GutoVeronezi** [comment]
@rohityadavcloud, done.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
awful lot of travis failures

**Comment by DaanHoogland** [comment]
re-try

**Comment by shwstppr** [comment]
@GutoVeronezi PR is marked for milestone 4.17.1.0, schema changes are also in 4170to4171 upgrade path but base branch is main. Should it be changed to 4.17 instead? cc @DaanHoogland

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Review by JoaoJandre — APPROVED** [review]
CLGTM

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql [hunk 1]
```diff
@@ -20,4 +20,110 @@
...
+        async_job.id job_id,
+        async_job.uuid job_uuid,
+        async_job.job_status job_status,
+        async_job.account_id job_account_id,
+        domain_router.template_version template_version,
+        domain_router.scripts_version scripts_version,
+        domain_router.is_redundant_router is_redundant_router,
+        domain_router.redundant_state redundant_state,
+        domain_router.stop_pending stop_pending,
+        domain_router.role role
```
Comment (shwstppr) [inline engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql:97]:
```suggestion
        domain_router.role role,
        domain_router.software_version software_version
```

Added with 4.17.0

**Review by shwstppr — CHANGES_REQUESTED** [review]
Missing a change in domeain_router_view that was added in 4.17.0. Getting error,

```
2022-08-02 08:39:31,178 ERROR [o.a.c.e.o.NetworkOrchestrator] (Work-Job-Executor-5:ctx-1e07947c job-36/job-37 ctx-3db22a03) (logid:5ef8342f) Caught exception: DB Exception on: com.mysql.cj.jdbc.ClientPreparedStatement: SELECT domain_router_view.id, domain_router_view.name, domain_router_view.account_id, domain_router_view.account_uuid, domain_router_view.account_name, domain_router_view.account_type, domain_router_view.domain_id, domain_router_view.domain_uuid, domain_router_view.domain_name, domain_router_view.domain_path, domain_router_view.state, domain_router_view.created, domain_router_view.removed, domain_router_view.instance_name, domain_router_view.pod_id, domain_router_view.pod_uuid, domain_router_view.data_center_id, domain_router_view.data_center_uuid, domain_router_view.data_center_name, domain_router_view.cluster_id, domain_router_view.dns1, domain_router_view.dns2, domain_router_view.ip6_dns1, domain_router_view.ip6_dns2, domain_router_view.host_id, domain_router_view.host_uuid, domain_router_view.host_name, domain_router_view.hypervisor_type, domain_router_view.template_id, domain_router_view.template_uuid, domain_router_view.service_offering_id, domain_router_view.service_offering_uuid, domain_router_view.service_offering_name, domain_router_view.vpc_id, domain_router_view.vpc_uuid, domain_router_view.vpc_name, domain_router_view.nic_id, domain_router_view.nic_uuid, domain_router_view.is_default_nic, domain_router_view.ip_address, domain_router_view.gateway, domain_router_view.netmask, domain_router_view.ip6_address, domain_router_view.ip6_gateway, domain_router_view.ip6_cidr, domain_router_view.mac_address, domain_router_view.broadcast_uri, domain_router_view.isolation_uri, domain_router_view.network_id, domain_router_view.network_uuid, domain_router_view.network_name, domain_router_view.network_domain, domain_router_view.traffic_type, domain_router_view.project_id, domain_router_view.project_uuid, domain_router_view.project_name, domain_router_view.job_id, domain_router_view.job_uuid, domain_router_view.job_status, domain_router_view.uuid, domain_router_view.template_version, domain_router_view.scripts_version, domain_router_view.redundant_state, domain_router_view.is_redundant_router, domain_router_view.guest_type, domain_router_view.role, domain_router_view.software_version FROM domain_router_view WHERE domain_router_view.id = 4  AND domain_router_view.removed IS NULL 
```

**Comment by shwstppr** [comment]
After fixing schema, tested both inter and intra-cluster storage migration for a VR and it worked successfully

File: engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql (hunk 1 above)
Comment (GutoVeronezi) [reply engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql]: Thanks, @shwstppr, just committed the fix.

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan test

**Review by shwstppr — APPROVED** [review]

**Comment by shwstppr** [comment]
re-kicking travis

**Comment by shwstppr** [comment]
@GutoVeronezi can you please check why tests from `test_vpc_routers` are failing on travis repeatedly

File: engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql [hunk 2]
```diff
@@ -20,4 +20,111 @@
...
+        data_center.ip6_dns2 ip6_dns2,
+        host.id host_id,
+        host.uuid host_uuid,
+        host.name host_name,
+        host.cluster_id cluster_id,
+        vm_template.id template_id,
+        vm_template.uuid template_uuid,
+        service_offering.id service_offering_id,
+        disk_offering.uuid service_offering_uuid,
+        disk_offering.name service_offering_name,
```
Comment (shwstppr) [inline engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql:66]:
```suggestion
        service_offering.uuid service_offering_uuid,
        service_offering.name service_offering_name,
```

**Review by shwstppr — CHANGES_REQUESTED** [review]
@GutoVeronezi this is causing failures

File: engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql [hunk 3]
```diff
@@ -20,4 +20,111 @@
...
+            left join
+        `cloud`.`data_center` ON vm_instance.data_center_id = data_center.id
+            left join
+        `cloud`.`host` ON vm_instance.host_id = host.id
+            left join
+        `cloud`.`vm_template` ON vm_instance.vm_template_id = vm_template.id
+            left join
+        `cloud`.`service_offering` ON vm_instance.service_offering_id = service_offering.id
+            left join
+        `cloud`.`disk_offering` ON vm_instance.service_offering_id = disk_offering.id
```
Comment (shwstppr) [inline engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql:120]: This isn't needed

File: engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql (hunk 2 above)
Comment (GutoVeronezi) [reply engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql]: Done. Many thanks @shwstppr.

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**